examples/templates/docker-devcontainer/main.tf

```diff
@@ -0,0 +1,372 @@
+  # changes in production.
+  version = "~> 1.0"
+}
+
+# See https://registry.coder.com/modules/coder/git-clone
+module "git-clone" {
+  count    = data.coder_workspace.me.start_count
+  source   = "registry.coder.com/coder/git-clone/coder"
+  agent_id = coder_agent.main.id
+  url      = data.coder_parameter.repo_url.value
+  base_dir = "~"
+  # This ensures that the latest non-breaking version of the module gets
+  # downloaded, you can also pin the module version to prevent breaking
+  # changes in production.
+  version = "~> 1.0"
+}
+
+# Automatically start the devcontainer for the workspace.
+resource "coder_devcontainer" "repo" {
+  count            = data.coder_workspace.me.start_count
+  agent_id         = coder_agent.main.id
+  workspace_folder = "~/${module.git-clone[0].folder_name}"
+}
+
+resource "docker_volume" "home_volume" {
+  name = "coder-${data.coder_workspace.me.id}-home"
+  # Protect the volume from being deleted due to changes in attributes.
+  lifecycle {
+    ignore_changes = all
+  }
+  # Add labels in Docker to keep track of orphan resources.
+  labels {
+    label = "coder.owner"
+    value = data.coder_workspace_owner.me.name
+  }
+  labels {
+    label = "coder.owner_id"
+    value = data.coder_workspace_owner.me.id
+  }
+  labels {
+    label = "coder.workspace_id"
+    value = data.coder_workspace.me.id
+  }
+  # This field becomes outdated if the workspace is renamed but can
+  # be useful for debugging or cleaning out dangling volumes.
+  labels {
+    label = "coder.workspace_name_at_creation"
+    value = data.coder_workspace.me.name
+  }
+}
+
+resource "docker_volume" "docker_volume" {
+  name = "coder-${data.coder_workspace.me.id}-docker"
+  # Protect the volume from being deleted due to changes in attributes.
+  lifecycle {
+    ignore_changes = all
+  }
+  # Add labels in Docker to keep track of orphan resources.
+  labels {
+    label = "coder.owner"
+    value = data.coder_workspace_owner.me.name
+  }
+  labels {
+    label = "coder.owner_id"
+    value = data.coder_workspace_owner.me.id
+  }
+  labels {
+    label = "coder.workspace_id"
+    value = data.coder_workspace.me.id
+  }
+  # This field becomes outdated if the workspace is renamed but can
+  # be useful for debugging or cleaning out dangling volumes.
+  labels {
+    label = "coder.workspace_name_at_creation"
+    value = data.coder_workspace.me.name
+  }
+}
+
+resource "docker_container" "workspace" {
+  count = data.coder_workspace.me.start_count
+  image = local.workspace_image
+
+  # NOTE: The `privileged` mode is one way to run Docker-in-Docker,
+  # which is required for the devcontainer to work. IF this is not
+  # desired, you can remove this line. However, you will need to ensure
+  # that the devcontainer can run Docker commands in some other way.
+  # Mounting the host Docker socket is one way to do this, but it is
+  # strongly discouraged because workspaces will then compete for
+  # control of the devcontainers.
```

Comment:
Suggest linking to https://coder.com/docs/admin/templates/extending-templates/docker-in-workspaces